### DISCUSSION TITLE
why remove langchain?

### DISCUSSION THREAD

**Discussion by griffith-h** [langgenius/dify#5702] [General]
Is there any brother who can give me some advice? I will go to the corresponding document or issue.（google translate）

**Comment by luzhongqiu** [comment]
langchian是真的丑陋的代码

**Reply by griffith-h** [reply]
害，已无那方面追求

**Comment by ZuzooVn** [comment]
> Is there any brother who can give me some advice? I will go to the corresponding document or issue.（google translate）

https://dify.ai/blog/the-model-runtime-restructure

**Reply by griffith-h** [reply]
very good bro, 3ks
